openverse_catalog/dags/providers/provider_api_scripts/inaturalist.py

```diff
@@ -1,111 +1,161 @@
-# In dbt, the schema would be created separately, and steps 1-5 would each be a "model"
-# and dbt would navigate the dependencies more or less automagically.
-LOAD_PROCESS = [
-    '00_create_schema.sql',
-    '01_photos.sql',
-    '02_observations.sql',
-    '03_taxa.sql',
-    '04_observers.sql',
-    '05_final_tsv.sql'
-]
-CONNECTION_ID = os.getenv("AIRFLOW_CONN_POSTGRES_OPENLEDGER_TESTING")
-
-
-def run_sql_file(file_name, file_path=SCRIPT_DIR, conn_id=CONNECTION_ID):
-    """
-    The process is really written in SQL so this script just enables logging 
-    and monitoring jobs, but this is the basic function to run the SQL files 
-    for each step.
-    """
-    logger.info(f"Running {file_name} using DB connection {conn_id}")
-    result = 'SQL failed. See log for details.'
-    try:
-        assert file_name[-4:]=='.sql'
-        assert os.path.exists(file_path + file_name)
-        db = psycopg2.connect(conn_id)
-        cursor = db.cursor()
-        query = dedent(open(file_path + file_name, 'r').read())
-        cursor.execute(query)
-        result = cursor.fetchall()
-        db.commit()
-        logger.info("Success!")
-    except Exception as e:
-        logger.warning(f"SQL step failed due to {e}")
-    return result
-
-
-def main():
-    """
-    This is really just looping through the SQL steps defined above, with some additional logging.
-    """
-    logger.info("Begin: Inaturalist script")
-    
-    for f in LOAD_PROCESS:
-        image_count = run_sql_file(f)
-        logger.info(f"Results: {str(image_count)}")
-    logger.info(f"Total images pulled: {image_count}")
-    logger.info('Terminated!')
-
-
-if __name__ == '__main__':
-    main()
+SCRIPT_DIR = (
+    "/usr/local/airflow/openverse_catalog/dags/providers/provider_csv_load_scripts/"
+    + "inaturalist/"
+)
+PG_TO_JSON_TEMPLATE = dedent(
+    open(os.path.join(SCRIPT_DIR, "05_export_dbpage_to_json_template.sql")).read()
+)
+
+
+class inaturalistDataIngester(ProviderDataIngester):
+
+    # TO DO: consider using this sql to get a max number of db pages to define a
+    # get_should_continue function
+    # select relpages
+    # from pg_class t join pg_namespace s on t.relnamespace=s.oid
+    # where s.nspname='inaturalist' and t.relname='photos';
+
+    providers = {"image": prov.INATURALIST_DEFAULT_PROVIDER}
+
+    def get_next_query_params(self, old_query_params=None, **kwargs):
+        """Page counter"""
+        if old_query_params is None:
+            return {"page_number": 0}
+        else:
+            next_page = old_query_params["page_number"] + 1
+            return {"page_number": next_page}
+
+    def get_response_json(self, query_params: Dict):
+        """
+        Call the SQL to pull json from Postgres, where the raw data has been loaded.
+        """
+        db_page_number = str(query_params["page_number"])
+        sql_string = PG_TO_JSON_TEMPLATE.replace("db_page_number", db_page_number)
+        pg = PostgresHook(POSTGRES_CONN_ID)
```

Comment:
This hook is initiated every time this function is called, which seems like a lot of overhead. Do you think we could override the `__init__` function on the class and add this as an instance variable which we can reuse where needed?

Reply:
🤦  Of course! Yes, I will absolutely do that.

Reply:
I think this is done [here](https://github.com/WordPress/openverse-catalog/blob/bf452b494466a2c9f1e7ade3bf986e12dce88607/openverse_catalog/dags/providers/provider_api_scripts/inaturalist.py#L59), yes?